Mailbox/DataLoader.cs

```diff
@@ -1,21 +1,58 @@
-﻿using System.Collections.Generic;
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
 using System.IO;
 
 namespace Mailbox
 {
-    public class DataLoader
+    public class DataLoader : IDisposable
     {
+        public Stream stream { get; set; }
```

Comment:
Properties should be PascalCase
```suggestion
        public Stream Stream { get; }
```